### PR TITLE
Slm 298 create task for monthly slm reporting

### PR DESCRIPTION
**Pull request by dhirajshettymoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeService.kt [hunk 1]
```diff
@@ -36,15 +47,34 @@ class BarcodeService(
...
+
+            // if no validation exception thrown above it means that the barcode is ready for delivery
+            trackScanEvent(ScanEventType.READY_FOR_DELIVERY, barcode, userId, location)
+          } catch (e: ValidationException) {
+            when (e.errorCode.code) {
+              "DUPLICATE" -> trackScanEvent(ScanEventType.DUPLICATE, barcode, userId, location)
+              "EXPIRED" -> trackScanEvent(ScanEventType.EXPIRED, barcode, userId, location)
+              "RANDOM_CHECK" -> trackScanEvent(ScanEventType.RANDOM_CHECK, barcode, userId, location)
+            }
+            throw e
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeService.kt:76]: `
trackScanEvent(ScanEventType.valueOf(e.errorCode.code), barcode, userId, location)
`
Would be better I think

**Review by AledWEvans — DISMISSED** [review]
LGTM, one comment your call

File: src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeService.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeService.kt:76]: Changed it initially but then decided to keep it the same as there are a few ValidationExceptions thrown and this is the safest code possible.

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeService.kt:74]: updated to a single when clause as suggested.

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**